File: python-algorithm/leetcode/problem_387.py
"""387. First Unique Character in a String
https://leetcode.com/problems/first-unique-character-in-a-string/

Given a string, find the first non-repeating character in it and
return its index. If it doesn't exist, return -1.

Examples:

s = "leetcode"
return 0.

s = "loveleetcode"
return 2.

Note: You may assume the string contains only lowercase English letters.
"""


class Solution:
    def first_uniq_char(self, s: str) -> int:
        store = {}
        for c in s:
            store[c] = 1 if c not in store else (1 + store[c])
        for i in range(len(s)):
            if store[s[i]] == 1:
                return i
        return -1

File: java-algo/src/main/java/com/leetcode/Problem387.java
package com.leetcode;

/**
 * 387. First Unique Character in a String
 * https://leetcode.com/problems/first-unique-character-in-a-string/
 */
class Problem387 {
    public int firstUniqChar(String s) {
        int[] counter = new int[26];
        for (int i = 0; i < s.length(); i++) {
            counter[s.charAt(i) - 'a']++;
        }
        for (int i = 0; i < s.length(); i++) {
            if (counter[s.charAt(i) - 'a'] == 1) return i;
        }
        return -1;
    }
}
